npm/npm_import.bzl

```diff
@@ -271,22 +273,54 @@ def npm_translate_lock(
 
         patches: A map of package names or package names with their version (e.g., "my-package" or "my-package@v1.2.3")
             to a label list of patches to apply to the downloaded npm package. Paths in the patch
-            file must start with `extract_tmp/package` where `package` is the top-level folder in
-            the archive on npm. If the version is left out of the package name, the patch will be
-            applied to every version of the npm package.
+            file must be relative to the root of the package. If the version is left out of the package name,
+            the patch will be applied to every version of the npm package.
+
+            For example,
+
+            ```
+            patches = {
+                "@foo/bar": ["//:patches/foo+bar.patch"],
+                "fum@0.0.1": ["//:patches/fum@0.0.1.patch"],
+            },
+            ```
+
+            List of patches List of public hoist packages are additive when there are multiple matches for a package.
 
         patch_args: A map of package names or package names with their version (e.g., "my-package" or "my-package@v1.2.3")
             to a label list arguments to pass to the patch tool. Defaults to -p0, but -p1 will
             usually be needed for patches generated by git. If patch args exists for a package
             as well as a package version, then the version-specific args will be appended to the args for the package.
 
+            ```
+            patch_args = {
+                "*": ["-p1"]
+                "@foo/bar": ["-p0"]
+                "fum@0.0.1": ["-p2"]
+            },
+            ```
+
+            List of patch args are not additive. The move specific name match takes precedence.
+
         custom_postinstalls: A map of package names or package names with their version (e.g., "my-package" or "my-package@v1.2.3")
             to a custom postinstall script to apply to the downloaded npm package after its lifecycle scripts runs.
             If the version is left out of the package name, the script will run on every version of the npm package. If
             a custom postinstall scripts exists for a package as well as for a specific version, the script for the versioned package
             will be appended with `&&` to the non-versioned package script.
 
-        prod: If true, only install dependencies.
+            For example,
+
+            ```
+            custom_postinstalls = {
+                "@foo/bar": "echo something > somewhere.txt",
+                "fum@0.0.1": "echo something_else > somewhere_else.txt",
+            },
+            ```
+
+            Custom postinstalls are additive and joined with ` && ` when there are multiple matches for a package.
+            More specific matches are appended to previous matches.
+
+        prod: If True, only install dependencies.
```

Comment:
add "but not devDependencies" otherwise it's not very clear. And putting in code quotes will also help explain this doesn't mean dependencies generally, it means that field in package.json

Reply:
✅